File: frontend/git-teller/models/scheduledReport/ScheduledReportJobListItemDTO.ts
export enum JobStatus {
    PENDING = "PENDING",
    RUNNING = "RUNNING",
    SUCCESS = "SUCCESS",
    FAILURE = "FAILURE",
}

export interface ScheduledReportJobListItemDTO {
    dataTo: string;
    dataFrom: string;
    scheduledFor: string;
    repoUri: string;
    status: JobStatus;
    retryCount: number;
}

File: backend/src/main/kotlin/pt/isel/model/scheduledReport/ScheduledReportJobListItemDTO.kt
package pt.isel.model.scheduledReport

import pt.isel.entity.report.model.JobStatus
import pt.isel.entity.report.schedule.ScheduledReportJobEntity
import java.time.Instant

data class ScheduledReportJobListItemDTO(
    val dataTo: Instant,
    val dataFrom: Instant,
    val scheduledFor: Instant,
    val repoUri: String,
    val status: JobStatus,
    val retryCount: Int,
){
    companion object{
        fun create(job: ScheduledReportJobEntity, repoUri: String) =
            ScheduledReportJobListItemDTO(
                job.dataTo, job.dataFrom,
                job.scheduledFor, repoUri,
                job.state.status,
                job.retryCount
            )
    }
}